homeassistant/components/pioneer/media_player.py

```diff
@@ -109,24 +115,49 @@ def telnet_request(cls, telnet, command, expected_prefix):
 
     def telnet_command(self, command):
         """Establish a telnet connection and sends command."""
-        try:
+        tries = MAX_TRIES
+        while tries > 0:
+            tries = tries - 1
             try:
-                telnet = telnetlib.Telnet(self._host, self._port, self._timeout)
-            except (ConnectionRefusedError, OSError):
-                _LOGGER.warning("Pioneer %s refused connection", self._name)
-                return
-            telnet.write(command.encode("ASCII") + b"\r")
-            telnet.read_very_eager()  # skip response
-            telnet.close()
-        except telnetlib.socket.timeout:
-            _LOGGER.debug("Pioneer %s command %s timed out", self._name, command)
+                try:
+                    telnet = telnetlib.Telnet(self._host, self._port, self._timeout)
+                    telnet.write(command.encode("ASCII") + b"\r")
+                    telnet.read_very_eager()  # skip response
+                    telnet.close()
+                    break
+                except (ConnectionRefusedError, OSError):
+                    _LOGGER.warning(
+                        "telnet_command: Pioneer %s refused connection", self._name
+                    )
+                    time.sleep(TRY_DELAY)
```

Comment:
As far as I know this is blocking IO and probably won't be accepted by core developers. Have a look at
https://developers.home-assistant.io/docs/asyncio_working_with_async

Reply:
I see why you don't want blocking I/O, but it's already blocking in it's current version.
This only changes the code to behave a little bit smarter in regards to connection failures (that happen on a regular basis after every single command send as i described previosly).
I dont' think I'll find the time to rewrite this to be non-blocking currently.